internal/api/phone.go

```diff
@@ -76,15 +79,12 @@ func (a *API) sendPhoneConfirmation(ctx context.Context, tx *storage.Connection,
 	if err != nil {
 		return "", internalServerError("error generating otp").WithInternalError(err)
 	}
-	*token = fmt.Sprintf("%x", sha256.Sum224([]byte(phone+otp)))
 
-	var message string
-	if config.Sms.Template == "" {
-		message = fmt.Sprintf(defaultSmsMessage, otp)
-	} else {
-		message = strings.Replace(config.Sms.Template, "{{ .Code }}", otp, -1)
+	message, err := parseSmsTemplate(config.Sms.Template, otp)
+	if err != nil {
+		return "", internalServerError("invalid sms template").WithInternalError(err)
 	}
-
+	*token = fmt.Sprintf("%x", sha256.Sum224([]byte(phone+otp)))
```

Comment:
Did we make a function for this hash?

Reply:
Not yet but we should, let me get to that